pallets/parachain-staking/src/weights.rs

```diff
@@ -68,57 +66,57 @@ pub trait WeightInfo {
 pub struct SubstrateWeight<T>(PhantomData<T>);
 impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	fn join_candidates() -> Weight {
-		(96_000_000 as Weight)
-			.saturating_add(T::DbWeight::get().reads(8 as Weight))
-			.saturating_add(T::DbWeight::get().writes(5 as Weight))
+		(277_000_000 as Weight)
```

Comment:
This base value is ~1_000 whatever `Config::MaxCollatorCandidates: Get<u32>` that we set. I set `MaxCollatorCandidates to `300` in the runtime now.  It could go higher but so would the `join_candidates` weight.

TODO:
- [ ] add sudo dispatchables for (re)setting `MaxCollatorCandidates` (make mutable storage item), test as well